lib/ansible/modules/storage/ceph/ceph_volume.py

```diff
@@ -0,0 +1,273 @@
+            - Will set the crush device class for the OSD.
+        required: false
+    dmcrypt:
+        description:
+            - If set to True the OSD will be encrypted with dmcrypt.
+        required: false
+        type: bool
+        default: false
+
+
+author:
+    - Andrew Schoen (@andrewschoen)
+'''
+
+EXAMPLES = '''
+- name: set up a filestore osd with an lv data and a journal partition
+  ceph_volume:
+    objectstore: filestore
+    data: data-lv
+    data_vg: data-vg
+    journal: /dev/sdc1
+
+- name: set up a bluestore osd with a raw device for data
+  ceph_volume:
+    objectstore: bluestore
+    data: /dev/sdc
+
+- name: set up a bluestore osd with an lv for data and partitions for block.wal and block.db
+  ceph_volume:
+    objectstore: bluestore
+    data: data-lv
+    data_vg: data-vg
+    db: /dev/sdc1
+    wal: /dev/sdc2
+'''
+
+RETURN = '''#  '''
+
+from ansible.module_utils.basic import AnsibleModule
+import datetime
+
+
+def get_data(data, data_vg):
+    if data_vg:
+        data = "{0}/{1}".format(data_vg, data)
+    return data
+
+
+def get_journal(journal, journal_vg):
+    if journal_vg:
+        journal = "{0}/{1}".format(journal_vg, journal)
+    return journal
+
+
+def get_db(db, db_vg):
+    if db_vg:
+        db = "{0}/{1}".format(db_vg, db)
+    return db
+
+
+def get_wal(wal, wal_vg):
+    if wal_vg:
+        wal = "{0}/{1}".format(wal_vg, wal)
+    return wal
+
+
+def run_module():
+    module_args = dict(
+        cluster=dict(type='str', required=False, default='ceph'),
+        subcommand=dict(type='str', required=False, default='lvm', choices=['lvm']),
+        objectstore=dict(type='str', required=True, choices=['bluestore', 'filestore']),
+        data=dict(type='str', required=True),
+        data_vg=dict(type='str', required=False),
+        journal=dict(type='str', required=False),
+        journal_vg=dict(type='str', required=False),
+        db=dict(type='str', required=False),
+        db_vg=dict(type='str', required=False),
+        wal=dict(type='str', required=False),
+        wal_vg=dict(type='str', required=False),
+        crush_device_class=dict(type='str', required=False),
+        dmcrypt=dict(type='bool', required=False, default=False),
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    cluster = module.params['cluster']
+    subcommand = module.params['subcommand']
+    objectstore = module.params['objectstore']
+    data = module.params['data']
+    data_vg = module.params.get('data_vg', None)
+    journal = module.params.get('journal', None)
+    journal_vg = module.params.get('journal_vg', None)
+    db = module.params.get('db', None)
+    db_vg = module.params.get('db_vg', None)
+    wal = module.params.get('wal', None)
+    wal_vg = module.params.get('wal_vg', None)
+    crush_device_class = module.params.get('crush_device_class', None)
+    dmcrypt = module.params['dmcrypt']
+
+    cmd = [
```

Comment:
Check if volume already exists or not before proceeding to the creation of volume.